Content.Server/ReagentOnItem/EntitySystems/ReagentOnItemSystem.cs

```diff
@@ -0,0 +1,91 @@
+    ///     It will apply all reagents that can stick and will spill anything
+    ///     that cannot stick to the floor.
+    /// </summary>
+    /// <returns> Returns false if there is no solution or if entity isn't an item and true otherwise. </returns>
+    public bool AddReagentToItem(EntityUid item, Solution reagentMixture)
+    {
+        if (reagentMixture.Volume <= 0 || !HasComp<ItemComponent>(item))
+        {
+            return false;
+        }
+
+        // Make sure the item doesn't have a non stick surface (E.g jani gloves or boots)
+        if (!HasComp<NonStickSurfaceComponent>(item))
+        {
+            // Yes this code is sussy but its much better than what was used before.
+            // All suspect code for this system is contained here (I hope).
+
+            // Remove all reagents that can actually stick to things and apply them
+            // to the item.
+            var volSpaceLube = reagentMixture.RemoveReagent("SpaceLube", reagentMixture.Volume).Double();
+            var volSpaceGlue = reagentMixture.RemoveReagent("SpaceGlue", reagentMixture.Volume).Double();
+
+            if (volSpaceLube > 0)
+            {
+                var lubed = EnsureComp<SpaceLubeOnItemComponent>(item);
+                AddAndDetermineExtra(lubed, "SpaceLube", volSpaceLube, reagentMixture);
+            }
+            if (volSpaceGlue > 0)
+            {
+                var glued = EnsureComp<SpaceGlueOnItemComponent>(item);
+                AddAndDetermineExtra(glued, "SpaceGlue", volSpaceGlue, reagentMixture);
+            }
+        }
+        else
+        {
+            // This means the item had a nonstick surface.
+            _popup.PopupEntity(Loc.GetString("non-stick-gloves-reagent-falls-off", ("target", Identity.Entity(item, EntityManager))), item);
+        }
+
+        // This spills the remaining mixture that wasn't applied.
+        _puddle.TrySpillAt(item, reagentMixture, out var puddle, false);
+
+        return true;
+
+    }
+    /// <summary>
+    ///     Actually add the (valid) reagent to the item and if there is any extra.
+    /// </summary>
+    /// <returns> The amount of leftover reagent that could not be applied. </returns>
+    private static void AddAndDetermineExtra(ReagentOnItemComponent comp, string reagentName, double volToAdd, Solution spillPool)
+    {
+        // This is the TOTAL amount of reagent on the item, old amount + stuff we are adding now
+        var total = comp.AmountOfReagentLeft + volToAdd;
+
+        // If the total is > capacity it will just put the maximum amount on the item.
+        // Ohterwise, its just the total amount!
+        comp.AmountOfReagentLeft = Math.Min(total, comp.ReagentCapacity);
+
+        // This means there is going to be extra reagents we need to add to the
+        // puddle.
+        if (total > comp.ReagentCapacity)
+        {
+            spillPool.AddReagent(reagentName, total - comp.ReagentCapacity);
+        }
+    }
```

Comment:
Rename `AmountOfReagentLeft` since it's no longer a reagent. Something like `EffectStacks`. (like FireStacks) Or maybe just turn into seconds left unless you want the decrementing to be separate from the time.

That'll make the code easier to read and understand. Reagent as part of the name confused me.

This includes renaming related methods like AddAndDetermineExtra to be clearer. Maybe ConvertReagentToStacks.

Reply:
That's a good idea! That function was hard to name 😆